core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/PathTrackingInputFormat.java

```diff
@@ -140,13 +143,19 @@ public NullWritable getCurrentKey() throws IOException, InterruptedException {
     protected abstract StructuredRecord getCurrentValue(StructuredRecord.Builder recordBuilder)
       throws IOException, InterruptedException;
 
+    protected abstract StructuredRecord getCurrentValueNull() throws IOException, InterruptedException;
```

Comment:
instead of this, reverse the order and change getCurrentValue to `StructuredRecord.Builder startCurrentValue()`. This class calls `startCurrentValue`, then sets the path field for it.